Scripts/Runtime/Entities/ManagedReference/ManagedReferenceStore.cs

```diff
@@ -27,6 +27,7 @@ static ManagedReferenceStore()
         [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterAssembliesLoaded)]
         private static void Init()
         {
+            s_IDProvider?.Dispose();
```

Comment:
@mbaker3 I noticed this when having a static IDProvider in `ResolveTargetUtil`. I believe this fixes a potential issue but wanted to flag it to you so you can weigh in.

Reply:
Today there's no real problem with forgetting to dispose. Stale instances remain listening to an unused instance of IDProvider and probably get garbage collected.

It's good practice though in case some day disposing an ID instance actually does become critical.

Reply:
Ah ok, cool